### PR TITLE
Set 'type' of external listener as required

### PR DESCRIPTION
**Pull request by stanlyDoge**
### Type of change
- Bugfix

### Description
The issue is descibed in https://github.com/strimzi/strimzi-kafka-operator/issues/872

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaListenerExternal.java [hunk 1]
```diff
@@ -38,7 +37,6 @@
             "* `route` type uses OpenShift Routes to expose Kafka." +
             "* `loadbalancer` type uses LoadBalancer type services to expose Kafka." +
             "* `nodeport` type uses NodePort type services to expose Kafka.")
-    @JsonIgnore
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaListenerExternal.java]: Do we need to set `@JsonProperty(required = true)`?

**Comment by ppatierno** [comment]
@stanlyDoge the #872 says that the same problem could be on `authentication` with related `type` field. Have you checked that one? At same time it mentions to fix crd-generator as well but maybe you should check with @tombentley for that.

**Comment by stanlyDoge** [comment]
I think the same problem is in the logging. Sadly this fix does not work. I am still investigating how to fix this.

**Comment by tombentley** [comment]
@stanlyDoge I opened 964 which makes the right annotation changes to avoid double including the `type` property.

**Comment by stanlyDoge** [comment]
Thanks for info @tombentley. After this change, the validation does not seem to work properly. Kafka cluster is configured with this setting without warning ```external: {
        "type" : unknown
      }``` 
And even if I add the required annotation to the `type` field, the generated CRD does not contain this information and therefore lets me use setting ```external: {}```.

**Comment by stanlyDoge** [comment]
Fixed by https://github.com/strimzi/strimzi-kafka-operator/pull/964

**Pull request closed without merging**